### PR TITLE
Default output timezone v3

### PR DESCRIPTION
**Pull request by markstory**
This pull request continues the work @ionas started in #8753. By having a default output timezone on the Time/Date classes we make it easier to localize all the datetime output an application generates by allowing datetime preferences to be centralized.

### CODE REVIEW

**Comment by ADmad** [comment]
Tests failing.

**Comment by markstory** [comment]
@ADmad Looks like more libicu woes. Those same tests pass fine for me locally with a newer version of libicu. I'll get them fixed though.

**Comment by davidyell** [comment]
Looking forward to this! Good work team :+1:

**Comment by ADmad** [comment]
@markstory :disappointed:  Tests pass for me locally too using "trusty".

File: tests/TestCase/I18n/TimeTest.php [hunk 1]
```diff
@@ -482,31 +544,96 @@ public function testI18nFormat($class)
...
     public function testI18nFormatWithOffsetTimezone($class)
     {
+        // Default output format is in UTC
         $time = new $class('2014-01-01T00:00:00+00');
         $result = $time->i18nFormat(\IntlDateFormatter::FULL);
         $expected = 'Wednesday January 1 2014 12:00:00 AM GMT';
         $this->assertTimeFormat($expected, $result);
 
+        $class::setDefaultOutputTimezone('GMT+09:00');
         $time = new $class('2014-01-01T00:00:00+09');
```
Comment (ionas) [inline tests/TestCase/I18n/TimeTest.php]: Well @markstory this was the key issue with my tests failing. I assumed that there should be a conversion of the offset to the output timezone. I don't use offsets personally and I am okay with this if you are but my gut tells me it should be different (either replacing the timezone or calculating the difference and applying that).

Thanks for making it work(tm)! :)

**Comment by ionas** [comment]
Looks good from my side. Don't forget to merge https://github.com/cakephp/app/pull/363 towards the same milestone, when this gets merged.

I can take care of the docs once its merged.

**Comment by lorenzo** [comment]
Good job @markstory @ionas

**Comment by dereuromark** [comment]
:tada: Awesome!

**Pull request merged**